### PR TITLE
SIS Import Data Modal

### PR DESCRIPTION
**Pull request by rbiz4**

### CODE REVIEW

**Comment by rbiz4** [comment]
For now, the modal shows up automatically, as soon as the advising page opens up, as below
image
My guess is that we want this to stop when there is actually data to render from SIS.

Currently, there are two ways to make the modal automatically show up upon opening the advising page:

Method one (what I am using right now): I call this.props.triggerSISImportDataModal() in top_bar_advising.jsx, as shown below:
image

Method two (I am not using this): I set the visibility to true in the modal reducer, as shown below:
image

This gives us two ways to control whether/not the modal shows up automatically upon opening the advisor page.

My question is, should the automatic pop-up control be something that the page itself is responsible for (Method one), or is it something that the Modal should be responsible for (Method two)

**Comment by miaboloix** [comment]
using just the toggle function is fine for now i think

File: static/js/redux/ui/modals/SIS_import_data_modal.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            </div>);
+        const modalBody =
+            (<div className="modal-body">
+                <h3>In order to share and connect to your
+                    official courses from SIS to Advisors,
+                    you must agree to import data from SIS
+                    (or something like this). Click the
+                    button below to be taken to the Consent
+                    Page on SIS.</h3>
+            </div>);
```
Comment (miaboloix) [inline static/js/redux/ui/modals/SIS_import_data_modal.jsx:50]: Instead, I would write something like: "In order to connect with your advisors through Semester.ly, you must agree to Semester.ly importing your data from SIS." And then have the "Import SIS" button.

File: static/js/redux/ui/semesterly.jsx [hunk 2]
```diff
@@ -159,7 +159,6 @@ class Semesterly extends React.Component {
         <IntegrationModalContainer />
         <TutModalContainer />
         <PeerModalContainer />
-        <SaveCalendarModalContainer />
         <FinalExamsModalContainer />
```
Comment (miaboloix) [inline static/js/redux/ui/semesterly.jsx:162]: Is this intentional?

File: static/js/redux/ui/top_bar_advising.jsx [hunk 3]
```diff
@@ -76,6 +79,27 @@ class TopBarAdvising extends React.Component {
...
+                    id="mock-btn-tooltip"
+                    class="tooltip"
+                    type="dark"
+                    place="bottom"
+                    effect="solid"
+                >
+                    <span>SIS Import Data Modal</span>
+                </ReactTooltip>
+            </div>
+        );
```
Comment (miaboloix) [inline static/js/redux/ui/top_bar_advising.jsx:102]: I think this is okay for now but add a TODO comment to remove this and instead add an "Import Data" button.  Then, create an Issue to track this TODO. Maybe this issue can be called "Replace star with an Import Data button that triggers the Import Data modal"

**Review by miaboloix — CHANGES_REQUESTED** [review]

File: static/js/redux/ui/semesterly.jsx (hunk 2 above)
Comment (rbiz4) [reply static/js/redux/ui/semesterly.jsx:162]: This was accidental, will add it back.

File: static/js/redux/ui/modals/SIS_import_data_modal.jsx (hunk 1 above)
Comment (rbiz4) [reply static/js/redux/ui/modals/SIS_import_data_modal.jsx:50]: Will revise

File: static/js/redux/ui/top_bar_advising.jsx (hunk 3 above)
Comment (rbiz4) [reply static/js/redux/ui/top_bar_advising.jsx:102]: Will change button

File: static/js/redux/ui/top_bar_advising.jsx [hunk 4]
```diff
@@ -30,6 +32,7 @@ class TopBarAdvising extends React.Component {
         this.comments_collapsed = 'neutral';
         this.toggleComments = this.toggleComments.bind(this);
         this.renderUserForPrint = this.renderUserForPrint.bind(this);
+        this.props.triggerSISImportDataModal()
```
Comment (JJamesWWang) [inline static/js/redux/ui/top_bar_advising.jsx:35]: Wouldn't this make it always pop up?

Comment (JJamesWWang) [reply static/js/redux/ui/top_bar_advising.jsx:35]: Okay, I saw your comment that it's like this for now. Maybe add a # TODO then

File: static/js/redux/ui/top_bar_advising.jsx (hunk 3 above)
Comment (jwfung) [reply static/js/redux/ui/top_bar_advising.jsx]: I noticed that Semesterly makes use of Fontawesome.com icons if you want to look into that, although I think just a button that says "import data" would be fine

**Review by miaboloix — APPROVED** [review]

**Review by miaboloix — CHANGES_REQUESTED** [review]
Pull most recent cs310 branch and also check for lint errors (npm run lint).

**Comment by Boccca2014** [comment]
Can you merge in the latest cs310 branch?

**Comment by miaboloix** [comment]
Lint errors!

**Comment by JJamesWWang** [comment]
I get a weird scrollbar when it first appears.
Scrollbar

**Comment by JJamesWWang** [comment]
The modal also doesn't appear when I go to /advising

**Review by JJamesWWang — CHANGES_REQUESTED** [review]
See above comments

**Comment by rbiz4** [comment]
I commented out the auto-pop-up part, and I left a TODO, linked to an issue (https://github.com/jhuopensource/semesterly/issues/219) about it

**Comment by rbiz4** [comment]
The scrollbar happens with other (pre-existing) modals as well
image

**Review by JJamesWWang — APPROVED** [review]

**Review by miaboloix — APPROVED** [review]
lgtm

**Pull request merged**